extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,215 @@
+        params,
+        query,
+      });
+    }
+    this._addConfiguration(config);
+  }
+
+  /**
+   * Get authorization headers for wado and qido calls
+   * @returns {object} xhrRequestHeaders
+   */
+  private getAuthorizationHeader(): object {
+    const xhrRequestHeaders = {};
+    const authHeaders = this.userAuthenticationService.getAuthorizationHeader();
+    if (authHeaders && authHeaders.Authorization) {
+      xhrRequestHeaders.Authorization = authHeaders.Authorization;
+    }
+    return xhrRequestHeaders;
+  }
+
+  /**
+   * Generates the header for wado messages for a specific client
+   * @param config
+   * @returns {object} wado Headers
+   */
+  private generateWadoHeader(config): object {
+    const authorizationHeader = this.getAuthorizationHeader();
+    //Generate accept header depending on config params
+    const formattedAcceptHeader = utils.generateAcceptHeader(
+      config.acceptHeader,
+      config.requestTransferSyntaxUID,
+      config.omitQuotationForMultipartRequest
+    );
+
+    return {
+      ...authorizationHeader,
+      Accept: formattedAcceptHeader,
+    };
+  }
+
+  /**
+   * Sets authorization headers for all clients before queries
+   * @returns {void}
+   */
+  public setQidoHeaders(): void {
+    this.clients.forEach(
+      client => (client.qidoDicomWebClient.headers = this.getAuthorizationHeader())
+    );
+  }
+
+  /**
+   * Sets wado headers for all clients before queries
+   * @returns {void}
+   */
+  public setWadoHeaders(): void {
+    this.clients.forEach(
+      client => (client.wadoDicomWebClient.headers = this.generateWadoHeader(client))
+    );
+  }
+
+  /**
+   * Sets authorization headers for wado clients before queries
+   * @returns {void}
+   */
+  public setAuthorizationHeadersForWADO(): void {
+    this.clients.forEach(
+      client => (client.wadoDicomWebClient.headers = this.getAuthorizationHeader())
+    );
+  }
+
+  /**
+   * Returns a function that returns if a client have reject abilities
+   * @returns {function} function that returns if a client can reject
+   */
+  public clientCanReject() {
+    return name => {
+      const client = this.clients.find(client => client.name === name);
+      return client?.supportsReject;
+    };
+  }
+
+  /**
+   * Returns reject function of a client
+   * @param name
+   * @returns {function} reject function
+   */
+  public getClientReject(name) {
+    const client = this.clients.find(client => client.name === name);
+    if (client?.supportsReject) {
+      return dcm4cheeReject(client.wadoRoot);
+    }
+  }
+
+  /**
+   * Returns the qido client
+   * @param name
+   * @returns {object} qido client
+   */
+  public getQidoClient(name = undefined): object {
+    if (this.clients.length) {
+      if (name) {
+        const client = this.clients.find(client => client.name === name);
+        return client?.qidoDicomWebClient;
+      } else {
+        return this.clients[0].qidoDicomWebClient;
+      }
+    }
+  }
```

Comment:
This can be written much better

```js
public getQidoClient(name?: string): object {
  if (!this.clients.length) {
    return;
  }

  const client = name ? this._getClinent(name) : this.clients[0];
  return client?.qidoDicomWebClient;
}

```